R/Figures.R

```diff
@@ -1138,6 +1138,385 @@ TADA_TwoCharacteristicScatterplot <- function(.data, id_cols = "TADA.ComparableD
+#' Scatterplot <- TADA_MultiScatterplot(Data_Nutrients_UT, id_cols = c("TADA.ComparableDataIdentifier", "MonitoringLocationName"), groups = c("UTAH LAKE 1 MI EAST OF PELICAN POINT", "Utah Lake 2 miles west of Vineyard"))
+#' Scatterplot[[1]]
+#' Scatterplot[[2]]
+#' 
+#' # Create multiple scatterplots and view the first plot in list.
+#' # In this example, we will group by both TADA.ComparableDataIdentifier
+#' # and MonitoringLocationTypeName (e.g. stream, reservoir, canal, etc.) and
+#' # because which MonitoringLocationTypeName was not specified, only the top 
+#' # 4 MonitoringLocationTypeName by field counts in the original dataset provided
+#' # will be included for each scatterplot generated.
+#' # Load example dataset:
+#' data(Data_Nutrients_UT)
+#' Scatterplot_output <- TADA_MultiScatterplot(Data_Nutrients_UT, id_cols = c("TADA.ComparableDataIdentifier", "MonitoringLocationTypeName"))
+#' # This example generates 14 scatterplots
+#' Scatterplot_output[[5]]
+#' Scatterplot_output[[8]]
+#' Scatterplot_output[[10]]
+#'
+#' # Load example dataset:
+#' data(Data_6Tribes_5y_Harmonized)
+#' # Filter the example data so it includes only one TADA.ComparableDataIdentifier
+#' df <- dplyr::filter(Data_6Tribes_5y_Harmonized, TADA.ComparableDataIdentifier %in% c("TOTAL PHOSPHORUS, MIXED FORMS_UNFILTERED_AS P_MG/L"))
+#' # Creates a scatterplot of the three specified sites of interest in the same plot.
+#' TADA_MultiScatterplot(df, id_cols = c("TADA.ComparableDataIdentifier", "MonitoringLocationName"), groups = c("Upper Red Lake: West", "Upper Red Lake: West-Central","Upper Red Lake: East Central"))
+#' # If no groups are selected, return the 4 groups (by MonitoringLocationName) with the greatest number of field count results for that TADA.ComparableDataIdentifier
+#' TADA_MultiScatterplot(df, id_cols = c("TADA.ComparableDataIdentifier", "MonitoringLocationName"))
+#' # Both commands below will return a single scatterplot for just the single comparable data identifier. 
+#' TADA_MultiScatterplot(df, id_cols = c("TADA.ComparableDataIdentifier", "TADA.ComparableDataIdentifier"))
+#' TADA_MultiScatterplot(df)
+#' 
+
+TADA_MultiScatterplot <- function(.data, id_cols = c("TADA.ComparableDataIdentifier"), groups = NULL) {
+  # check .data is data.frame
+  TADA_CheckType(.data, "data.frame", "Input object")
+  
+  # check .data has required columns
+  TADA_CheckColumns(.data, id_cols)
+  
+  # check .data has required columns
+  reqcols <- c(
+    "TADA.ResultMeasureValue",
+    "TADA.ResultMeasure.MeasureUnitCode",
+    "ActivityStartDate"
+  )
+  
+  # check .data has required columns
+  TADA_CheckColumns(.data, reqcols)
+  
+  # if left blank, ensure comparable data identifier is in the id_cols vector
+  if (is.null(id_cols) | length(unique(id_cols)) == 1) {
+    id_cols <- c("TADA.ComparableDataIdentifier","TADA.ComparableDataIdentifier")
+  }
+  id_cols <- c("TADA.ComparableDataIdentifier", vecsets::vsetdiff(id_cols, "TADA.ComparableDataIdentifier"))
```

Comment:
Do you think it would be possible to use a function from a different package (one already used in EPATADA) here? Is there a way that "setdiff" could be used here instead? We already have quite a few dependencies and if this is the only place where this package is used, an alternative would be preferable if possible.

Should users be able to provide an id_cols input that does not include TADA.ComparableDataIdentifier as one of the inputs? If they should not be able to, this section can probably be simplified as users would only have to supply one id_col like MonitoringLocationIdentifier or WaterbodyTypeName.

If TADA.ComparableDataIdentifier should not be a required/predetermined id_cols, what are some examples of other groups of columns that might be used?

Reply:
This should be possible. I was going to ask what are your thoughts with adding functions from a different package and if too many dependencies of other packages become problematic. It was used to make sure TADA.ComparableDataIdentifier was the first items in a vector, but if we no longer make this a user input for an argument in id_cols we might not even need vsetdiff or an alternative way to handle.

I prefer the idea of a single id_cols argument. I think I was trying to keep it consistent with some of the other plot functions that were using c('TADA.ComparableDataIdentifier', 'another field').

Would OrganizationFormalName be a good example of column groupings?